sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -175,6 +178,23 @@ abstract class OffsetWindowFunctionFrameBase(
     }
   }
 
+  override def prepare(rows: ExternalAppendOnlyUnsafeRowArray): Unit = {
+    resetStates(rows)
```

Comment:
so the fix is always call `resetStates`?

Reply:
I improved the behavior. only `FrameLessOffsetWindowFunctionFrame` need `resetStates` here.